File: backend/src/main/java/com/bonifert/backend/model/Topic.java
package com.bonifert.backend.model;

import com.bonifert.backend.model.user.UserEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Topic {
  @Id
  @GeneratedValue
  private Long id;
  private String name;
  @ManyToOne
  private UserEntity userEntity;
  @CreationTimestamp
  private LocalDateTime createdAt;
  @UpdateTimestamp
  private LocalDateTime modifiedAt;
  @OneToMany(mappedBy = "topic")
  private List<Term> terms;
}

File: frontend/src/components/types/Topic.ts
export interface Topic {
  name: string;
  id: number;
  terms: Term[];
  createdAt: string;
  modifiedAt: string;
  priority: string;
}

export interface Term {
  id: number;
  name: string;
  definition: string;
  nextShowDateTime: string;
}